src/main/java/ui/UI.java

```diff
@@ -544,4 +552,36 @@ public HWND getMainWindowHandle() {
         return mainWindowHandle;
     }
 
+    public void showNotificationPane(Node graphic, String text, Action action) {
+        hideNotificationPane();
+        Platform.runLater(() -> {
```

Comment:
What happens with this line and the previous is something like this:

```java
void showNotificationPanel(...) {
    Platform.runLater(() -> {
        // hide
    });
    Platform.runLater(() -> {
        // show
    });
}
```

The `show` block is guaranteed to follow the `hide` block in a sequential environment, but if `showNotificationPanel` is called from many threads concurrently, the `show` and `hide` blocks may interleave. Any permutation where `hideN` precedes `showN` is possible:

```
hide1 show1 hide2 show2 // normal
hide2 show2 hide1 show1 // also okay
hide1 hide2 show1 show2 // unexpected things may happen
hide2 hide1 show1 show2 // this too
```
I've been staring at it for a while and I don't *think* there is incorrect behaviour right now, but there might be in future. :grimacing: Just to be on the safe side we should ensure that the hiding, then showing is done as one atomic action.

Another issue is that if there are multiple concurrent calls, order between them is undefined and only the last will appear. We should buffer the notification requests somehow... but that's probably for another issue.